src/CMakeLists.txt

```diff
@@ -5,8 +5,14 @@ project(minetest)
 INCLUDE(CheckIncludeFiles)
 INCLUDE(CheckLibraryExists)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+	set(COMPILER_ABI_FLAG "-Wabi=11")
+else()
+	set(COMPILER_ABI_FLAG "-Wabi")
+endif()
+
 # Add custom SemiDebug build mode
-set(CMAKE_CXX_FLAGS_SEMIDEBUG "-O1 -g -Wall -Wabi" CACHE STRING
+set(CMAKE_CXX_FLAGS_SEMIDEBUG "-O1 -g -Wall ${COMPILER_ABI_FLAG}" CACHE STRING
```

Comment:
`CMAKE_CXX_FLAGS_SEMIDEBUG` is overwritten later anyway so this can be left as "-O1 -g -Wall" (or even empty?)